lib/connection.js

```diff
@@ -159,13 +163,32 @@ var Connection = module.exports = function(options) {
   }
 
   var cacheOptions = {
-    key: function(type) { return type ? "describe." + type : "describe"; }
+    key: function(type) { 
+      return type 
+        ? type.type ? "describe." + type.type : "describe." + type
+        : "describe"; 
+    }
   };
   this.describe$ = this.cache.makeCacheable(this.describe, this, cacheOptions);
   this.describe = this.cache.makeResponseCacheable(this.describe, this, cacheOptions);
   this.describeSObject$ = this.describe$;
   this.describeSObject = this.describe;
 
+  var batchCacheOptions = {
+    key: function(options) {
+      var types = options.types;
+      var autofetch = options.autofetch || false;
+      var typesToFetch = types.length > MAX_BATCH_REQUESTS 
+        ? (autofetch ? types : types.slice(0, MAX_BATCH_REQUESTS))
+        : types;
+      var keys = [];
+      typesToFetch.forEach(function (type) { keys.push('describe.' + type); });
+      return keys;
+    }
+  };
+  this.batchDescribe = this.cache.makeResponseCacheable(this.batchDescribe, this, batchCacheOptions);
+  // this.batchDescribeSObjects = this.batchDescribe;
```

Comment:
Why is this commented out? Should it be removed?

Reply:
User error. I uncommented it in `fix(conn): accidentally commented out line of code`.

Reply:
NIce catch @amphro